src/modules/loops/iter_loop.rs

```diff
@@ -73,26 +73,56 @@ impl SyntaxModule<ParserMetadata> for IterLoop {
 impl TranslateModule for IterLoop {
     fn translate(&self, meta: &mut TranslateMetadata) -> String {
         let name = &self.iter_name;
-        let iterable = self.iterable.translate(meta);
+        let expr = self.iter_expr.translate(meta);
         match self.iter_index.as_ref() {
             Some(index) => {
                 // Create an indentation for the index increment
                 meta.increase_indent();
                 let indent = meta.gen_indent();
                 meta.decrease_indent();
-                [format!("{index}=0;"),
-                    format!("for {name} in {iterable}"),
-                    "do".to_string(),
-                    self.block.translate(meta),
-                    format!("{indent}(( {index}++ )) || true"),
-                    "done".to_string()].join("\n")
+                if let Some(conditional) = self.iter_expr.conditional(name) {
+                    [
+                        format!("{index}=0;"),
+                        format!("for {name} in {expr}"),
+                        "do".to_string(),
+                        format!("if {conditional}"),
+                        "then".to_string(),
+                        self.block.translate(meta),
+                        format!("{indent}(( {index}++ )) || true"),
+                        "fi".to_string(),
+                        "done".to_string(),
+                    ].join("\n")
+                } else {
+                    [
+                        format!("{index}=0;"),
+                        format!("for {name} in {expr}"),
+                        "do".to_string(),
+                        self.block.translate(meta),
+                        format!("{indent}(( {index}++ )) || true"),
+                        "done".to_string(),
+                    ].join("\n")
+                }
             },
             None => {
-                [format!("for {name} in {iterable}"),
-                    "do".to_string(),
-                    self.block.translate(meta),
-                    "done".to_string()].join("\n")
-            }
+                if let Some(conditional) = self.iter_expr.conditional(name) {
+                    [
+                        format!("for {name} in {expr}"),
+                        "do".to_string(),
+                        format!("if {conditional}"),
```

Comment:
Please note, this will only work with string literals:
```
#!/usr/bin/env amber

echo "[works]"
loop index, item in glob("*.md") {
    echo "{index}: {item}"
}

echo "[fails]"
let wildcard = "*.md"
loop index, item in glob(wildcard) {
    echo "{index}: {item}"
}
```
I do not see any way around this (short of calling `find`, which has its own problems) because Bash glob expansion does not work with quoted globs.